Sources/MessagingPush/PushHandling/iOSPushEventListener.swift

```diff
@@ -3,94 +3,32 @@ import CioTracking
 import Foundation
 
 @available(iOSApplicationExtension, unavailable)
-/**
-
- # Why is this class a singleton?
- 1. Class stores data that needs to be kept in-memory.
-
- ---
-
- # Why is this class not stored in the digraph?
-
- This class is the SDK's `UNUserNotificationCenterDelegate` instance.  Meaning, the CIO SDK registers this class with the OS in order to
- receive callbacks when push notifications are interacted with.
-
- It's important that the instance of this class provided to the OS stays in memory so our SDK can receive those OS callbacks.
-
- In order to promise this class's singleton instance stays in memory, the singleton instance is *not* stored inside of the digraph (like all other singletons in our SDK is).
-
- This is a workaound to prevent this scenario:
- - native iOS SDK is initialized.
- - PushEventListener singleton instance is created.
- - PushEventListener singleton instance is registered with the OS to receive push notification callbacks.
- - SDK wrappers re-initialize the native iOS SDK when the SDK wrapper SDK is initialized.
- - During the native iOS SDK's initialization, the SDK's digraph instance is re-created. All objects (and singletons) in that old digraph instance are deleted from memory.
- - That's bad! If the PushEventListener singleton instance was stored in the digraph, it would be deleted from memory. The OS would no longer be able to send push notification callbacks to the SDK.
- */
+// sourcery: InjectRegister = "PushEventHandler"
 class iOSPushEventListener: PushEventHandler {
-    // Singleton instance of this class maintained outside of the digraph.
-    public static let shared = iOSPushEventListener()
-
-    // Instances of all dependencies that automated tests can override.
-    private var overrideJsonAdapter: JsonAdapter?
-    private var overrideModuleConfig: MessagingPushConfigOptions?
-    private var overridePushClickHandler: PushClickHandler?
-    private var overridePushHistory: PushHistory?
-
-    // Below is a set of getters for all dependencies of this class.
-    // Each getter will first check if a test override exists. If so, return that. Otherwise, return an instance from the digraph.
-    // It's important that we use do not keep a strong reference to any dependencies or to the digraph. Otherwise, the SDK would crash if the SDK's digraph gets re-initialized and
-    // this class tries to access old instances of dependencies.
-    private var jsonAdapter: JsonAdapter? {
-        overrideJsonAdapter ?? diGraph?.jsonAdapter
-    }
```

Comment:
All of this code that enabled this class to be created and maintained in-memory outside of the digraph still is required, but has been moved into `Sources/MessagingPush/UserNotificationsFramework/UserNotificationsFrameworkAdapter.swift` because that is the SDK's new `UNUserNotificationCenterDelegate` instance. 

This class used to be the delegate, but got changed after we de-coupled this class from the `UserNotifications` framework.